interfaces/mount/backend.go

```diff
@@ -61,6 +61,9 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 	if err != nil {
 		return fmt.Errorf("cannot obtain mount security snippets for snap %q: %s", snapName, err)
 	}
+	if snapInfo.InstanceKey != "" {
```

Comment:
I would make this unconditional and handle the nothing-to-do-case internally. It is in line with the layout call below (not all snaps use layouts)